src/components/network-map-tab.js

```diff
@@ -209,16 +221,127 @@ export const NetworkMapTab = ({
         []
     );
 
-    const loadMapGeoData = useCallback(() => {
+    const getEquipmentsNotFoundIds = useCallback(
+        (foundEquipmentPositions, allEquipments) => {
+            const foundEquipmentsIds = Array.from(
+                foundEquipmentPositions.keys()
+            );
+
+            const notFoundEquipmentsIds = allEquipments
+                .filter(
+                    (s) =>
+                        !foundEquipmentsIds.includes(s.id) ||
+                        temporaryGeoData.temporaryGeoDataIds.has(s.id)
+                )
+                .map((s) => s.id);
+
+            return notFoundEquipmentsIds;
+        },
+        [temporaryGeoData.temporaryGeoDataIds]
+    );
+
+    const getMissingEquipmentsPositions = useCallback(
+        (notFoundEquipmentsIds, fetchEquipmentCB) => {
+            if (notFoundEquipmentsIds.length === 0) {
+                return Promise.resolve([]);
+            }
+
+            return fetchEquipmentCB(
+                studyUuid,
+                currentNode?.id,
+                notFoundEquipmentsIds
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
+    const loadMissingGeoData = useCallback(() => {
+        const notFoundSubstationIds = getEquipmentsNotFoundIds(
+            geoData.substationPositionsById,
+            mapEquipments.getSubstations()
+        );
+        temporaryGeoData.addGeoDataIds(notFoundSubstationIds);
```

Comment:
when we add IDs, we always create a new set in TemporaryGeoData. But here we always readd what we already added, so it creates a new set everytime. We should avoid this ?

Reply:
Yes. It works with a simple Set::add(value: T)